applications: nrf_desktop: Power Manager documentation (#2102)
NordicPlayground/fw-nrfconnect-nrf: applications/nrf_desktop/doc/main.rst

.. _nrf_desktop_main:

Main module
###########

The main module is the entry point of the nRF Desktop application.

Module Events
*************

+------------------+-------------+--------------------------+------------------------+------------------+
| Source Module    | Input Event | This Module              | Output Event           | Sink Module      |
+==================+=============+==========================+========================+==================+
|                  |             |  :ref:`nrf_desktop_main` | ``module_state_event`` |                  |
+------------------+-------------+--------------------------+------------------------+------------------+

Configuration
*************

The module has no configuration options.

Implementation details
**********************

> The :ref:`nrf_desktop_main` module initializes the :ref:`event_manager` and then submits the module state event to inform the other modules that they can continue the application startup.

Reviewer: Should we change it to `module_state_event` for consistency?

Author: Done

.. _nrf_desktop_main:

Main module
###########

> The main module is the entry point of the application.

Reviewer 2: ```suggestion
The main module is the entry point of the nRF Desktop application.
```

Module Events
*************

+------------------+-------------+--------------------------+------------------------+------------------+
| Source Module    | Input Event | This Module              | Output Event           | Sink Module      |
+==================+=============+==========================+========================+==================+

> |                  |             |  :ref:`nrf_desktop_main` | ``module_state_event`` |                  |

Author: @MarekPieta - discussed offline. I think this is required for completeness but what we can do is to create an intermediate page that would hide source and sink modules for event where the number of these is large.

> |                  |             |  :ref:`nrf_desktop_main` | ``module_state_event`` |                  |

Author: Hi @greg-fer , it seems inconsistent at the moment so we need to decide on something and fix it everywhere. @MarekPieta will generate these tables using the script so we don't have to manually maintain them. Going back to the your original comment I would prefer to have a link as then the format would be the same as for source and sink.

> |                  |             |  :ref:`nrf_desktop_main` | ``module_state_event`` |                  |

Reviewer 2: I'd really prefer not to link to the page you're on. It's very confusing. If the script cannot handle it, then I'd prefer to discuss this more.

> |                  |             |  :ref:`nrf_desktop_main` | ``module_state_event`` |                  |

Reviewer: The script can decorate module name under `This module` column without `ref`.
In case the module appears also as one of Source/Sink modules it will be decorated with `ref` (for consistency with other modules in the column).
Will this be ok for you ?

+------------------+-------------+--------------------------+------------------------+------------------+

Configuration
*************

The module has no configuration options.

Implementation details
**********************

> The :ref:`nrf_desktop_main` module initializes the :ref:`event_manager` and then submits the module state event to inform the other modules that the main module is ready.

Author: My intention was to refer to type of the event but I know what you mean and can give name here.
As for module references - personally I would prefer to five references as it would be easier to navigate through documentation. We can fix it when we go and add prefix to all module names.

> The :ref:`nrf_desktop_main` module initializes the :ref:`event_manager` and then submits the module state event to inform the other modules that the main module is ready.
> The other modules continue the application startup from this point.

Reviewer 2: ```suggestion
The :ref:`nrf_desktop_main` module initializes the :ref:`event_manager` and then submits the module state event to inform the other modules that they can continue the application startup.
```
